### DISCUSSION TITLE
Why is Ocelot requesting for 1-2 minutes using the https protocol in an IIS environment?

### DISCUSSION THREAD

**Discussion by songlin-q** [ThreeMammals/Ocelot#1765] [Q&A] [answered]
Ocelot in IIS environment, using https protocol for request, request 6-7 times, it will be lagged, each time the waiting time is about 1-2 minutes

**Comment by raman-m** [comment]
:grin:
You are not alone! See: [IIS issues](/search?q=repo%3AThreeMammals%2FOcelot+IIS&type=issues)

**Comment by raman-m** [comment]
Hi Q!
Thanks for your interest in Ocelot!

I would say, Ocelot is not optimized for [IIS deployments](https://ocelot.readthedocs.io/en/latest/search.html?q=IIS&check_keywords=yes&area=default) and it is not recommended to deploy Ocelot app to IIS. Also, we have some [IIS Gotchas](../blob/develop/docs/introduction/gotchas.rst). 😸 
Such kind of custom environment requires a special ASP.NET MVC pipeline settings to run successfully. Each version of .NET framework requires special setup of ASP.NET MVC pipeline to [support IIS](https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/iis/?view=aspnetcore-7.0). You haven't described your IIS version and .NET version and Ocelot version!
To be fair, IIS has a vast and unstable pipeline, which is based on IIS modules and handlers, and which is better optimized for ASP.NET Web apps with frontends (aka Razor, Blazor), not just ASP.NET Wen API apps. Yes, we can host Web API apps in IIS environment but this scenario has overheads.
I would recommend you to deploy to Kestrel web server (self-hosting), Docker or public cloud providers like Azure.

Tell me please,
**Why do you use IIS for Ocelot app deployments?**

**Comment by raman-m** [comment, marked as answer]
Regarding slow response from IIS...
> Ocelot in IIS environment, using https protocol for request, request 6-7 times, it will be lagged, **each time the waiting time is about 1-2 minutes**

It seems, developer from #1657 had the same problem of slow responses. The problem could be wrong usage of InProcess setup for IIS. So, [Out-of-process hosting model](1657#issuecomment-1593578877) must be used for IIS deployments!!!

Read these docs:
- Review PR #1657 to understand the problem
- Read MS Learn docs to understand how to setup Ocelot app for IIS deployments: [Out-of-process hosting with IIS and ASP.NET Core | Microsoft Learn](https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/iis/out-of-process-hosting?view=aspnetcore-7.0)

**Comment by songlin-q** [comment]
Thank you for your reply
